pkg/stanza/docs/operators/syslog_parser.md

```diff
@@ -14,6 +14,7 @@ The `syslog_parser` operator parses the string-type field selected by `parse_fro
 | `protocol`                           | required         | The protocol to parse the syslog messages as. Options are `rfc3164` and `rfc5424`. |
 | `location`                           | `UTC`            | The geographic location (timezone) to use when parsing the timestamp (Syslog RFC 3164 only). The available locations depend on the local IANA Time Zone database. [This page](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) contains many examples, such as `America/New_York`. |
 | `enable_octet_counting`              | `false`          | Wether or not to enable [RFC 6587](https://www.rfc-editor.org/rfc/rfc6587#section-3.4.1) Octet Counting on syslog parsing (Syslog RFC 5424 only).  |
+| `allow_skip_pri_header`              | `false`          | Allow parsing records without the pri header. If the header is missing the message will be parsed and the priority and facility field will be set to their zero values. By default records without pri header will throw an exception. |
```

Comment:
> If the header is missing the message will be parsed and the priority and facility field will be set to their zero values.

Hm now that I'm thinking about it, this means that a Syslog message without PRI header and a message with PRI equal to `<0>` are parsed into the same log record. The only thing that tells them apart is the body, which is hard to automatically check. I think the user should be able to tell these two cases apart. It doesn't seem right that a Syslog message without PRI header results in a log record that has `Severity` set to `Fatal`.

Perhaps the `Severity` and the `priority` and `facility` attributes should not be set at all when the PRI header is missing? I wonder what you think.

Reply:
priority and facility are int so their default value is 0 if they are not set. I can check if the body starts with < and if not I can omit the facility and priority from the dictionary. 
that being said. 

-  this may cause an issue if someone is relying on those values to be in the dict (which seems to me like a fair assumption.) so this may cause more serious issues.
- this will fix RFC5424 and 3164. the more tricky case is when octec counting is enable. skipping the pri is not supported (for now) in this case anyway so this shouldn't be an issue (I do need to update the doc to specify this)